### PR TITLE
Add Kubebuilder tags to Gatewayclassconfig proto messages

### PR DESCRIPTION
**Pull request by sarahalsmiller**
### Description
- Added in kubebuiler tags to keep generated spec in line with previous CRD definition.
- Deleted fields that were in the documentation, but not in the original CRD definition for consistency

### Testing & Reproduction steps
- Linter passes, CI passes

### Links

https://github.com/hashicorp/consul-k8s/blob/main/control-plane/api/v1alpha1/api_gateway_types.go Original CRD definition. 

### PR Checklist

* [ ] updated test coverage
* [ ] external facing docs updated
* [ X ] appropriate backport labels added
* [ X ] not a security concern

### CODE REVIEW

File: proto-public/pbmesh/v2beta1/gatewayclassconfig.proto [hunk 1]
```diff
@@ -6,53 +6,52 @@ syntax = "proto3";
...
+  // +kubebuilder:validation:Minimum=1
+  // Minimum allowed number of gateway instances
   optional uint32 min_instances = 2;
+  // +kubebuilder:default:=8
+  // +kubebuilder:validation:Maximum=8
+  // +kubebuilder:validation:Minimum=1
+  // Max allowed number of gateway instances
   optional uint32 max_instances = 3;
+  // Resources defines the resource requirements for the gateway.
+  repeated k8s.io.api.core.v1.ResourceRequirements tolerations = 4;
```
Comment (jm96441n) [inline proto-public/pbmesh/v2beta1/gatewayclassconfig.proto:55]: should this be `resources` rather than `tolerations`?

**Review by NiniOak — COMMENTED** [review]
How do I test this change manually?

**Comment by sarahalsmiller** [comment]
> How do I test this change manually?

This is a struct change with no business logic so in theory all you would need to do is make sure the updated fields looks appropriate and the binary builds, but if you wanted to test to ensure you can build locally you can run following steps. 

```make proto```

Ensure the generated proto doesn't error

and

```make lint```
To make sure the generate proto code was generated correctly

**Review by NiniOak — APPROVED** [review]
No errors from `make proto` or `make lint` so approving. 🚀

**Review by jm96441n — APPROVED** [review]
LGTM!

**Pull request merged**